pkg/ruler/ruler.go

```diff
@@ -560,7 +566,7 @@ func (r *Ruler) getShardedRules(ctx context.Context, userID string) ([]*rules.Ru
 		return nil, fmt.Errorf("unable to inject user ID into grpc request, %v", err)
 	}
 
-	rgs := []*rules.RuleGroupDesc{}
+	rgs := make([]*rules.RuleGroupDesc, 0, len(rulers.Ingesters))
```

Comment:
I don't think you need this change. The number of Rulers in the ring does not correlate to the number of rule groups being evaluated.

Reply:
> The number of Rulers in the ring does not correlate to the number of rule groups being evaluated.

Yes, but it should be no more than `len(rulers.Ingesters)`. This fix is for avoid realloc slice.
if the slice is empty (before my fix), the len and cap is `0`. After each append, the len increase 1 but cap will be 0, 2, 2, 4, 4..., which make malloc many times.

Reply:
Can you add a comment for the same? I'll merge it after!

Reply:
I've added the comment.